scripts/build_arbor.sh

```diff
@@ -30,13 +31,29 @@ fi
 # remove old build files
 mkdir -p "$arb_build_path"
 
+# build modcc
+if [ "$ns_arb_modcc" == "ON" ]; then
+    mkdir -p "$modcc_build_path"
+
+    # build external modcc
+    msg "ARBOR: build modcc"
+    cd "$modcc_build_path"
+    cmake .. -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ >> "$out" 2>&1
+    make -j $ns_makej modcc >> "$out" 2>&1
+    cd ..
+fi
+
 # configure the build with cmake
 cd "$arb_build_path"
 cmake_args=-DCMAKE_INSTALL_PREFIX:PATH="$ns_install_path"
 cmake_args="$cmake_args -DARB_WITH_MPI=$ns_with_mpi"
 cmake_args="$cmake_args -DARB_WITH_GPU=$ns_arb_with_gpu"
 cmake_args="$cmake_args -DARB_ARCH=$ns_arb_arch"
 cmake_args="$cmake_args -DARB_VECTORIZE=$ns_arb_vectorize"
+if [ "$ns_arb_modcc" == "ON" ]; then
```

Comment:
The name `ns_arb_no_xcompile_modcc` is more descriptive than `ns_arb_modcc`.

The variable is not set by default.
It could be default initialised to `ns_arb_no_xcompile_modcc=OFF` with the other `ns_arb_*` variables in `scripts/environment.sh`, then overriden in system configuration scripts.

Reply:
Slightly changed to `ns_arb_xcompile_modcc`.